# Análisis Demográfico de Ingresos y Educación en Adultos: Exploración de Factores Socioeconómicos en Diversos Países

In [1]:
#Importamos la librería pandas
import pandas as pd 

You can download dataset here: 👇👇

https://github.com/WLozanoH/projects_freecodecamp/blob/main/projects/adult.data.csv

In [2]:
#cargamos el archivo 'csv'

df = pd.read_csv('adult.data.csv')
df.head()

age         workclass  fnlwgt  education  education-num  \
0   39         State-gov   77516  Bachelors             13   
1   50  Self-emp-not-inc   83311  Bachelors             13   
2   38           Private  215646    HS-grad              9   
3   53           Private  234721       11th              7   
4   28           Private  338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country salary  
0          2174             0              40  United-States  <=50K  
1             0             0              13  United-States  <=50K  
2             0             0              40  United-States  <=50K  
3             0             0              40  United-States  <=50K  
4             0             0              40           Cuba  <=50K

In [3]:
df.info() #no hay valores nulos

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32561 entries, 0 to 32560
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   age             32561 non-null  int64 
 1   workclass       32561 non-null  object
 2   fnlwgt          32561 non-null  int64 
 3   education       32561 non-null  object
 4   education-num   32561 non-null  int64 
 5   marital-status  32561 non-null  object
 6   occupation      32561 non-null  object
 7   relationship    32561 non-null  object
 8   race            32561 non-null  object
 9   sex             32561 non-null  object
 10  capital-gain    32561 non-null  int64 
 11  capital-loss    32561 non-null  int64 
 12  hours-per-week  32561 non-null  int64 
 13  native-country  32561 non-null  object
 14  salary          32561 non-null  object
dtypes: int64(6), object(9)
memory usage: 3.7+ MB


# Answer to question 📉📚

📌 * Pregunta 1: ¿Cuántas personas de cada raza están representadas en este set de datos?

In [4]:
race_count = df['race'].value_counts()
for  race in race_count.index:
    print(f"race: '{race}' - count: {race_count[race]}")



race: 'White' - count: 27816
race: 'Black' - count: 3124
race: 'Asian-Pac-Islander' - count: 1039
race: 'Amer-Indian-Eskimo' - count: 311
race: 'Other' - count: 271


🎯 Respuesta 1: 

Las razas de las personas en este set de datos están distribuídas de la siguiente manera:

        race: 'White' - count: 27816
        race: 'Black' - count: 3124
        race: 'Asian-Pac-Islander' - count: 1039
        race: 'Amer-Indian-Eskimo' - count: 311
        race: 'Other' - count: 271



📌 Pregunta 2: ¿Cuál es la edad promedio de los hombres?

In [5]:
promedio_edad_hombres = df.loc[df['sex']=='Male']['age'].mean()
promedio_edad_hombres = round(promedio_edad_hombres)
print(f'la edad promedio de los hombres es de: {promedio_edad_hombres} años.')

la edad promedio de los hombres es de: 39 años.


🎯 Respuesta 2: 

        la edad promedio de los hombres en este set de datos es de: 39 años.

        

* Pregunta 3: ¿Cuál es el porcentaje de personas que tienen un grado de licenciatura (Bachelor's degree)?


In [6]:
# paso 1: filtrar personas con 'education' == 'Bachelors'
degree_bachelor = df[df['education']== 'Bachelors']

# paso 2: hallar el total de personas con 'education' = 'Bachelors'
total_degree_bachelor = len(degree_bachelor)

# paso 3: hallar el total de personas con educacion.
count_df = len(df)
count_df

# paso4: hallar el porcentaje de personas que tienen licenciatura 'Bachelors' en este set de datos

porcentaje_bachelor = (total_degree_bachelor/count_df)*100
porcentaje_bachelor = round(porcentaje_bachelor,2)

print(f"En este set de datos:\n\n  El porcentaje de personas  con Educación = 'Bachelor' es: {porcentaje_bachelor}%.")


En este set de datos:

  El porcentaje de personas  con Educación = 'Bachelor' es: 16.45%.


🎯  Respuesta 3: 

        El porcentaje de personas  con Educación = 'Bachelor' es: 16.45%.

* Pregunta 4: ¿Qué porcentaje de personas con una educación avanzada (Bachelors, Masters o Doctorate) ganan más de 50k?

In [7]:
#paso 1: filtrar datos por educacion avanzada, educacion = (['Bachelors','Masters', 'Doctorate'])
education_advance = df[df['education'].isin(['Bachelors','Masters', 'Doctorate'])]

#paso 2: hallar el total de personas con  educacion avanzada
total_count_advance = len(education_advance)

#paso 3: filtrar las personas que ganan mas de 50K, con educacion avanzada
high_salary = education_advance[education_advance['salary']== '>50K']

#paso 4:  contar el total de personas que ganan mas de 50K, con educacion avanzada
count_high_salary = len(high_salary)

#paso 5: porcentaje  de personas que ganan mas de 50K, con educacion avanzada
porcentaje = (count_high_salary/total_count_advance) *100
porcentaje = round(porcentaje,2)

print(f"El porcentaje de personas con una educación avanzada (Bachelors, Masters o Doctorate) ganan más de 50k es:\n\nporcentaje: {porcentaje}%.")

El porcentaje de personas con una educación avanzada (Bachelors, Masters o Doctorate) ganan más de 50k es:

porcentaje: 46.54%.


🎯 Respuesta 4:

El porcentaje de personas con una educación avanzada (Bachelors, Masters o Doctorate) ganan más de 50k es:

        porcentaje: 46.54%.

📌 Pregunta 5: ¿Qué porcentaje de personas sin una educación avanzada generan más de 50k?

In [8]:
#paso 1: filtrar datos sin educacion avanzada, educacion = (['Bachelors','Masters', 'Doctorate'])
sin_education_advance = df[~df['education'].isin(['Bachelors','Masters', 'Doctorate'])]

#paso 2: hallar el total de personas sin  educacion avanzada
total_count_sin_advance = len(sin_education_advance)

#paso 3: filtrar las personas que ganan mas de 50K, sin educacion avanzada
high_salary = sin_education_advance[sin_education_advance['salary']== '>50K']

#paso 4:  contar el total de personas que ganan mas de 50K, sin educacion avanzada
count_high_salary = len(high_salary)

#paso 5: porcentaje  de personas que ganan mas de 50K, sin educacion avanzada
porcentaje = (count_high_salary/total_count_sin_advance) *100
porcentaje = round(porcentaje,2)

print(f"El porcentaje de personas sin una educación avanzada (Bachelors, Masters o Doctorate) ganan más de 50k es:\n\nporcentaje: {porcentaje}%.")

El porcentaje de personas sin una educación avanzada (Bachelors, Masters o Doctorate) ganan más de 50k es:

porcentaje: 17.37%.


🎯 Respuesta 5: 

    El porcentaje de personas sin una educación avanzada (Bachelors, Masters o Doctorate) ganan más de 50k es:

        porcentaje: 17.37%.

📌 Pregunta 6: ¿Cuál es el mínimo número de horas que una persona trabaja por semana?

In [9]:
min_hours = df['hours-per-week'].min()
print(f"El mínimo de horas trabajadas por semana es: {min_hours} hora.")


El mínimo de horas trabajadas por semana es: 1 hora.


🎯  Respuesta 6: 

        El minimo de horas trabajadas por semana es: 1 hora.

📌 Pregunta 7: ¿Qué porcentaje de personas que trabajan el mínimo de horas por semana tiene un salario de más de 50k?

In [10]:
#paso 1: hallar el minimo de horas trabajadas por semana
min_hours = df['hours-per-week'].min()

#paso 2: filtrar las personas que trabajan el minimo de horas
min_hours_workers = df[df['hours-per-week']== min_hours]

#paso 3: calcular el total de personas que trabajan el minimo de horas
total_min_hours_workers = len(min_hours_workers)

#paso 4: hallar el total de personas que ganan  mas de 50K y que trabajan el minimo de horas
min_hours_high_salary = len(min_hours_workers[min_hours_workers['salary'] == '>50K'])

#paso 5: hallar el porcentaje  de personas que ganan mas de 50K y que trabajan el minimo de horas
porcentaje = (min_hours_high_salary/total_min_hours_workers)*100
porcentaje = round(porcentaje,2)

print(f"El porcentaje de personas que ganan mas de 50K y trabajan un minimo de horas a la semana es:\n\n  Porcentaje: {porcentaje}%.")



El porcentaje de personas que ganan mas de 50K y trabajan un minimo de horas a la semana es:

  Porcentaje: 10.0%.


🎯 Respuesta 7: 

El porcentaje de personas que ganan mas de 50K y trabajan un minimo de horas a la semana es:

    Porcentaje: 10.0%.     

* Pregunta 8: ¿Qué país tiene el porcentaje más alto de personas que ganan >50k y cuál es ese porcentaje?

In [11]:
# paso 1: agrupamos por pais y contamos por salario con value_counts()
salary_counts = df.groupby('native-country')['salary'].value_counts()

#paso 2: lo convertimos en un data frame con unstack()
salary_counts = salary_counts.unstack()

#paso 3: calculamos el porcentaje

#calculamos el salario por país
salary_counts['total'] = salary_counts['<=50K'] + salary_counts['>50K']
salary_counts['>50K'] =  (salary_counts['>50K'] / salary_counts['total']) * 100

# paso 4: calculamos el porcentaje del salario por país mayor a 50k
#filtramos el pais por el índice
country_max_percentage = salary_counts['>50K'].idxmax()

#porcentaje mas alto de ese país:
max_percentage = salary_counts['>50K'].max()
max_percentage = round(max_percentage,2)

print(F"País con el porcentaje de personas con salario mayor a 50K\n\n País: {country_max_percentage}\n Porcentaje: {max_percentage}%.")


País con el porcentaje de personas con salario mayor a 50K

 País: Iran
 Porcentaje: 41.86%.


🎯 Respuesta 8:

País con el porcentaje de personas con salario mayor a 50K

        País: Iran
        Porcentaje: 41.86%.

📌 Pregunta 9: Identifica la ocupación más popular de aquellos que ganan >50k en India.

In [12]:
# Paso 1: Filtrar data por país = 'India' y 'salary' = '>50K'
filtered_df = df[(df['native-country'] == 'India') & (df['salary'] == '>50K')]

# Paso 2: Hallar la ocupacion mas popular con el filtrado anterior
occupation_popular = filtered_df['occupation'].value_counts()

# Paso 3: Determinar la ocupacion mas popular
most_popular_occupation = occupation_popular.idxmax()
count_popular_occupation = occupation_popular.max()

print(f"La ocupacion más popular en la india de aquellas personas que ganan más de 50K es:\n\n Ocupacion: {most_popular_occupation}\n Count: {count_popular_occupation}")

La ocupacion más popular en la india de aquellas personas que ganan más de 50K es:

 Ocupacion: Prof-specialty
 Count: 25


🎯 Respuesta 9: 

La ocupacion más popular en la india de aquellas personas que ganan más de 50K es:

    Ocupacion: Prof-specialty
    Count: 25

✏️  Conclusiones: 

Este análisis demográfico proporciona información valiosa a partir del dataset descargado, con preguntas clave que destacan aspectos importantes relacionados con raza, educación, salario y ocupación. A continuación, se resumen los puntos más importantes:

1. *Distribución racial:*

        La mayor representación en el conjunto de datos corresponde a personas identificadas como blancas, seguidas por negros y asiático-pacíficos. La menor representación es de personas clasificadas como "Other".

2. *Edad promedio de los hombres:* 

        La edad promedio de los hombres en este dataset es de 39 años.

3. *Porcentaje de personas con educación de licenciatura:*

        Aproximadamente el 16.45% de las personas tienen una licenciatura.

4. *Educación avanzada y salario mayor a 50k:* 

        Casi la mitad (46.54%) de las personas con un título avanzado (licenciatura, maestría o doctorado) ganan más de 50k.

5. *Personas sin educación avanzada y salario mayor a 50k:*

        Solo el 17.37% de las personas sin educación avanzada ganan más de 50k.

6. *Mínimo de horas trabajadas por semana:*

        La mínima cantidad de horas trabajadas por semana es 1 hora.

7. *Personas que trabajan el mínimo de horas y ganan más de 50k:* 

        Solo el 10% de las personas que trabajan el mínimo de horas ganan más de 50k.

8. *País con mayor porcentaje de personas que ganan >50k:* 

        Irán tiene el porcentaje más alto, con un 41.86% de personas que ganan más de 50k.

9. *Ocupación más popular entre los que ganan >50k en India:*

        La ocupación más común es "Prof-specialty", con 25 personas en esta categoría.

Este análisis resalta cómo ciertas variables, como la educación y la ocupación, influyen significativamente en los ingresos, y cómo los datos sociodemográficos varían entre diferentes grupos dentro del conjunto de datos.